lib/modules/manager/npm/post-update/pnpm.ts

```diff
@@ -66,16 +68,37 @@ export async function generateLockFile(
       extraEnv.NPM_AUTH = env.NPM_AUTH;
       extraEnv.NPM_EMAIL = env.NPM_EMAIL;
     }
-    const commands: string[] = [];
 
-    cmd = 'pnpm';
-    let args = 'install --recursive --lockfile-only';
+    let cmdOptions = '--recursive --lockfile-only';
     if (!GlobalConfig.get('allowScripts') || config.ignoreScripts) {
-      args += ' --ignore-scripts';
-      args += ' --ignore-pnpmfile';
+      cmdOptions += ' --ignore-scripts';
+      cmdOptions += ' --ignore-pnpmfile';
+    }
+    logger.trace({ cmdOptions }, 'pnpm command options');
+
+    if (
+      upgrades.length === 0 ||
```

Comment:
I think this should never happen. renovate should not call update artifacts when there are no updates

Reply:
it happens in the test suite, should I modify the inputs? 
I didn't feel comfortable doing so because I don't know the reel format it can have

Reply:
we should early return if `upgrades.length === 0` as there is nothing todo then